Frameworks/UIKit/UIImage.mm

```diff
@@ -44,487 +44,416 @@ @interface UIImage ()
-        [layer setContentsCenter:stretch];
-    }
+void UIImageSetLayerContents(CALayer *layer, UIImage *image) {
+  CGImageRef cgImage = (CGImageRef)[image CGImage];
+  CGImageRef curImage = (CGImageRef)[layer contents];
+
+  if (cgImage != curImage) {
+    [layer setContents:(id)cgImage];
+  }
+  if ([layer contentsScale] != [image scale]) {
+    [layer setContentsScale:[image scale]];
+  }
+  if ([layer contentsOrientation] != [image imageOrientation]) {
+    [layer setContentsOrientation:[image imageOrientation]];
+  }
+  CGRect stretch = [image _imageStretch];
+  if (!CGRectEqualToRect([layer contentsCenter], stretch)) {
+    [layer setContentsCenter:stretch];
+  }
 }
 
 @implementation UIImage {
-    CGImageRef m_pImage;
-    id _cacheName;
-    idretain _cacheImage;
-    CGRect _imageStretch;
-    UIEdgeInsets _imageInsets;
-    float _scale;
-    UIImageOrientation _orientation;
-    bool _isFromCache;
-    uint8_t* out;
-    uint8_t** row_pointers;
-    StrongId<NSData> _deferredImageData;
+  CGImageRef m_pImage;
+  id _cacheName;
+  idretain _cacheImage;
+  CGRect _imageStretch;
+  UIEdgeInsets _imageInsets;
+  float _scale;
+  UIImageOrientation _orientation;
+  bool _isFromCache;
+  uint8_t *out;
+  uint8_t **row_pointers;
+  StrongId<NSData> _deferredImageData;
 }
 
 /**
  @Status Stub
 */
 + (BOOL)supportsSecureCoding {
-    UNIMPLEMENTED();
-    return StubReturn();
+  UNIMPLEMENTED();
+  return StubReturn();
 }
 
 /**
  @Status Stub
 */
-- (instancetype)initWithCoder:(NSCoder*)decoder {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (instancetype)initWithCoder:(NSCoder *)decoder {
+  UNIMPLEMENTED();
+  return StubReturn();
 }
 
 /**
  @Status Stub
 */
-- (void)encodeWithCoder:(NSCoder*)encoder {
-    UNIMPLEMENTED();
+- (void)encodeWithCoder:(NSCoder *)encoder {
+  UNIMPLEMENTED();
 }
 
-+ (UIImageCachedObject*)cacheImage:(UIImage*)image withName:(NSString*)name {
-    //  Cache the image
-    UIImageCachedObject* obj = [UIImageCachedObject new];
-    obj->_scale = image->_scale;
-    obj->_imageStretch = image->_imageStretch;
-    obj->m_pImage = image->m_pImage;
-    CGImageRetain(obj->m_pImage);
-    obj->_cacheName = [name copy];
-    pthread_mutex_lock(&imageCacheLock);
-    [(id)g_imageCache setObject:(id)obj forKey:obj->_cacheName];
-    pthread_mutex_unlock(&imageCacheLock);
++ (UIImageCachedObject *)cacheImage:(UIImage *)image withName:(NSString *)name {
+  //  Cache the image
+  UIImageCachedObject *obj = [UIImageCachedObject new];
+  obj->_scale = image->_scale;
+  obj->_imageStretch = image->_imageStretch;
+  obj->m_pImage = image->m_pImage;
+  CGImageRetain(obj->m_pImage);
+  obj->_cacheName = [name copy];
+  pthread_mutex_lock(&imageCacheLock);
+  [(id)g_imageCache setObject:(id)obj forKey:obj->_cacheName];
+  pthread_mutex_unlock(&imageCacheLock);
+
+  return [obj autorelease];
+}
 
-    return [obj autorelease];
+static inline CGImageRef getImage(UIImage *self) {
```

Comment:
it's static inline.

---
In reply to: [84153721](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84153721) [](ancestors = 84153721)